pyccel/parser/semantic.py

```diff
@@ -2362,9 +2362,20 @@ def _visit_Return(self, expr, **settings):
             if not (isinstance(r, PyccelSymbol) and r == v):
                 a = Assign(v, r)
                 a.set_fst(expr.fst)
+                a = self._visit_Assign(a)
+                ind = a.rhs.get_attribute_nodes(IndexedElement)
+                if len(ind) == 0 and not isinstance(a.rhs, IndexedElement):
+                    a.lhs.is_temp = True
+                elif isinstance(a.rhs, IndexedElement) and a.rhs.base.is_argument:
+                    a.lhs.is_temp = True
+                elif len(ind) > 0:
+                    a.lhs.is_temp = True
+                    for b in ind:
+                        if not b.base.is_argument:
+                            a.lhs.is_temp = False
+                            break
```

Comment:
This section sets the `is_temp` property. Can't we add this property to `PyccelSymbol` and then it will be known from creation?

Reply:
i think adding this property to `PyccelSymbol` will cause modifications in many places, so it better to keep it in `Variable`, since it already has this kind of properties unlike `PyccelSymbol`

Reply:
why is it going to cause modifications in many places?

Reply:
I meant if we add it now we wont use it, since we only count on `Variable` object, and as i said its better to keep `PyccelSymbol` clean, and add this kind of attributes to `Variable` since it had a lot of them, unless you have some critical enhancement that requires `is_temp` in `PyccelSymbol`.